spec/util/app-service-mock.js

```diff
@@ -4,10 +4,47 @@ var util = require("util");
 var instance = null;
 
 function MockAppService() {
+    let self = this;
+
+    this.app = {
+        post: function(path, handler) {
+            // Assume that the first post handler will be for the /link path
```

Comment:
I would not assume this, and instead just check if the `path === "/_matrix/whatever/the/link/path/is`.